UI/platform-osx.mm

```diff
@@ -255,6 +255,51 @@ bool ProcessIsRosettaTranslated()
 #endif
 }
 
+@interface DockView : NSView {
+@private
+	QIcon icon;
+}
+@end
+
+@implementation DockView
+- (id)initWithIcon:(QIcon)icon
+{
+	self = [super init];
+	self->icon = icon;
+	return self;
+}
+- (void)drawRect:(NSRect)dirtyRect
+{
+	CGSize size = dirtyRect.size;
```

Comment:
@gxalpha How often is this called? If the icon (including the status icon) only changes on actual changes of the app's status, then redrawing the overlay on every draw call of the composer seems wasteful to me.

Reply:
It has to be called manually with `[tile display]`, so it's only called when `TaskbarOverlaySetIcon` is.